src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -183,11 +185,33 @@ public void testForInvalidCheckImplementation() throws Exception {
 
     @Test
     public void testProcessNonJavaFiles() throws Exception {
+        final List<String> lines =
+            new ArrayList<>(Arrays.asList("package com.puppycrawl.tools.checkstyle;", "",
+                "error public class InputTreeWalkerFileWithViolation {}"));
+
         final TreeWalker treeWalker = new TreeWalker();
-        treeWalker.setTabWidth(1);
+        final PackageObjectFactory factory = new PackageObjectFactory(
+            new HashSet<>(), Thread.currentThread().getContextClassLoader());
+        treeWalker.setModuleFactory(factory);
         treeWalker.configure(new DefaultConfiguration("default config"));
-        final File file = new File("src/main/resources/checkstyle_packages.xml");
-        treeWalker.processFiltered(file, new ArrayList<>());
+        final DefaultConfiguration childConfig = createCheckConfig(JavadocParagraphCheck.class);
+        treeWalker.setupChild(childConfig);
+
+        try {
+            treeWalker.processFiltered(new File("input.java"), lines);
+            fail("Exception expected");
+        }
+        catch (CheckstyleException ex) {
+            assertEquals("Invalid exception message",
+                "MismatchedTokenException occurred during the analysis of file input.java.",
+                ex.getMessage());
+        }
+        try {
+            treeWalker.processFiltered(new File("input.txt"), lines);
+        }
+        catch (CheckstyleException ex) {
+            fail("Exception is not expected, file with txt extension should not be processed");
```

Comment:
We said before catching an exception and just failing is bad practice.
Exception should be populated to junit runner to display to user.